server/src/test/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinatorTest.java

```diff
@@ -3275,4 +3275,49 @@ public void testSegmentIdShouldNotBeReallocated() throws IOException
     );
     Assert.assertNull(coordinator.retrieveSegmentForId(theId.asSegmentId().toString(), true));
   }
+
+  @Test
+  public void testRetrieveUnusedSegmentsForExactIntervalAndVersion() throws Exception
+  {
+    DataSegment unusedForDifferentVersion = createSegment(
+        Intervals.of("2024/2025"),
```

Comment:
Nit: use an interval which is easier to use in a name. You may even assign this interval value to a field named `Interval may2024` so that you can reuse it in multiple places.
```suggestion
        Intervals.of("2024-05/P1M"),
```